platforms/kubernetes/actions/deployments/rollout.ts

```diff
@@ -2,27 +2,109 @@ import { revisionRoute } from "../../common/knative/route.ts";
 import { upsertObject } from "../../common/objects.ts";
 import { AppContext } from "../../mod.ts";
 import { Namespace } from "../sites/create.ts";
+import { k8s } from "../../deps.ts";
+import { logger } from "deco/observability/otel/config.ts";
+import {
+  PLURAL_REVISIONS,
+  PLURAL_ROUTES,
+  GROUP_SERVINGKNATIVEDEV,
+  VERSION_V1,
+} from "../../constants.ts";
 
 export interface Props {
   site: string;
   deploymentId: string;
 }
 
+export interface MinPodsProps {
+  k8sApi: k8s.CustomObjectsApi;
+  site: string;
+  oldRevision: string;
+}
+
+export interface UpdateRevision {
+  ctx: AppContext;
+  site: string;
+  oldRevision: string;
+}
+
 export const Routes = {
   prod: (site: string) => `sites-${site}`,
   preview: (site: string, deploymentId: string) =>
     `${Routes.prod(site)}-${deploymentId}`,
 };
 
+const getCurrentProdRevision = async ({
+  k8sApi,
+  site,
+}: {
+  k8sApi: k8s.CustomObjectsApi;
+  site: string;
+}) => {
+  const currentProdRevision = await k8sApi
+    .getNamespacedCustomObject(
+      GROUP_SERVINGKNATIVEDEV,
+      VERSION_V1,
+      Namespace.forSite(site),
+      PLURAL_ROUTES,
+      Routes.prod(site)
+    )
+    .catch((err) => {
+      logger.error(err);
+    });
```

Comment:
you're logging it here but what happens then? the currentProdRevision will be undefined and that's it? Is there any action that we should take?